web/src/engine/websites/RawLazy.ts

```diff
@@ -70,25 +93,32 @@ export default class extends DecoratableMangaScraper {
 
     public override async FetchPages(chapter: Chapter): Promise<Page[]> {
         const chapterUrl = new URL(chapter.Identifier, this.URI).href;
-        let currentUrl = await this.FindNextStageUrl(chapterUrl, '.go-to-B');
-        currentUrl = await this.FindNextStageUrl(currentUrl, '.go-to-C');
+        let currentUrl = await FetchWindowScript<string>(new Request(chapterUrl), this.Script('.go-to-B'), 1500);
+        currentUrl = await FetchWindowScript<string>(new Request(currentUrl), IframeScript, 1500);
         const pages = await FetchCSS<HTMLImageElement>(new Request(currentUrl), 'div.z_content img');
         return pages.map(image => new Page(this, chapter, new URL(image.src)));
     }
 
-    private async FindNextStageUrl(currentUrl: string, selector: string): Promise<string> {
-        return FetchWindowScript<string>(new Request(currentUrl), this.Script(selector), 500);
-    }
-
     private Script(selector: string): string {
         return `
-            new Promise(resolve => {
+            new Promise( (resolve, reject) => {
+                const start = Date.now();
                 const interval = setInterval(function () {
-                    if (document.querySelector('${selector}').href) != '#') {
+                    try {
+                        if (document.querySelector('${selector}')?.href != '#') {
+                            clearInterval(interval);
+                            resolve(document.querySelector('${selector}').href);
+                        }
+                    } catch (error) {
                         clearInterval(interval);
-                        resolve(document.querySelector('${selector}').href));
+                        reject(error);
+                    } finally {
+                        if(Date.now() - start > 10_000) {
+                            clearInterval(interval);
+                            reject(new Error('Unable to get pictures after more than 10 seconds !'));
```

Comment:
This is probably shown to the user on failed download.
Use a more appropriate and localised message e.g., `FetchProvider_FetchWindow_TimeoutError`.

Reply:
Yup. How i am supposed to use localised message in injected promises? 😔

Reply:
Consider a different approach which will provide an appropriate error message
```javascript
new Promise((resolve, reject) => {
    const interval = setInterval(() => {
        console.log('Do some stuff ...');
    }, 1000);

    AbortSignal.timeout(10_000).onabort = function({ target: { reason } }) {
        clearInterval(interval);
        reject(reason);
    }
});
```